CMakeLists.txt

```diff
@@ -41,15 +64,168 @@ SET(PKG_CONFIG_EXTRA "plugindir=${PLUGINDIR}")
+  include/dynamic-graph/signal-base.h
+  include/dynamic-graph/signal-ptr.h
+  include/dynamic-graph/signal-time-dependent.h
+  include/dynamic-graph/signal-ptr.t.cpp
+  include/dynamic-graph/signal.t.cpp
+  include/dynamic-graph/time-dependency.h
+  include/dynamic-graph/time-dependency.t.cpp
+  include/dynamic-graph/signal-caster.h
+  include/dynamic-graph/signal-cast-helper.h
+  include/dynamic-graph/all-signals.h
+  include/dynamic-graph/signal-helper.h
+  include/dynamic-graph/entity-helper.h
+
+  include/dynamic-graph/tracer.h
+  include/dynamic-graph/tracer-real-time.h
+
+  include/dynamic-graph/command.h
+  include/dynamic-graph/eigen-io.h
+  include/dynamic-graph/linear-algebra.h
+  include/dynamic-graph/value.h
+
+  include/dynamic-graph/command-setter.h
+  include/dynamic-graph/command-setter.t.cpp
+  include/dynamic-graph/command-getter.h
+  include/dynamic-graph/command-getter.t.cpp
+  include/dynamic-graph/command-direct-getter.h
+  include/dynamic-graph/command-direct-setter.h
+  include/dynamic-graph/command-bind.h
+  include/dynamic-graph/all-commands.h
+
+  include/dynamic-graph/logger.h
+)
+
+SET(PUBLIC_HEADER ${${PROJECT_NAME}_HEADERS})
+
+INSTALL(TARGETS ${PROJECT_NAME}
+  EXPORT ${TARGETS_EXPORT_NAME}
+  PUBLIC_HEADER
+  INCLUDES DESTINATION include
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  )
+
+# We need to include Eigen3 because the CMakeFile at 16.04 LTS
+# is not so modern.
+target_include_directories(${PROJECT_NAME}
+  PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_BUILD_DIR}/include>
+   $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
+   $<BUILD_INTERFACE:${EIGEN3_INCLUDE_DIR}>
+   INTERFACE ${EIGEN3_INCLUDE_DIR}
+   $<INSTALL_INTERFACE:include>
+   )
+
+SET_TARGET_PROPERTIES(${LIBRARY_NAME} PROPERTIES SOVERSION ${PROJECT_VERSION})
+
+IF (UNIX)
+  TARGET_LINK_LIBRARIES(${LIBRARY_NAME} ${CMAKE_DL_LIBS} pthread)
+ENDIF (UNIX)
+
+TARGET_LINK_LIBRARIES(${LIBRARY_NAME} ${Boost_LIBRARIES})
+
+####################################
+### Plugins
+####################################
+SET(plugins_list
+        src/traces/tracer
+        src/traces/tracer-real-time
+)
+
+SET(tracer-real-time_dependency tracer)
+
+FOREACH(plugin_file ${plugins_list})
+  GET_FILENAME_COMPONENT(plugin ${plugin_file} NAME)
+  ADD_LIBRARY(${plugin} SHARED "${plugin_file}.cpp")
+
+  TARGET_LINK_LIBRARIES(${plugin}
+    ${PROJECT_NAME} ${${plugin}_dependency}
+    ${Boost_LIBRARIES})
+
+  SET_TARGET_PROPERTIES(${plugin}
+    PROPERTIES
+    PREFIX ""
+    )
+
+  INSTALL(TARGETS ${plugin}
+    EXPORT ${TARGET_NAME}
+    DESTINATION ${PLUGINDIR})
+ENDFOREACH(plugin_file)
+
 ADD_SUBDIRECTORY(include/${PROJECT_NAME})
 ADD_SUBDIRECTORY(tests)
 
-SETUP_PROJECT_FINALIZE()
+
+SETUP_PROJECT_PACKAGE_FINALIZE()
+
+get_cmake_property(_variableNames VARIABLES)
+list (SORT _variableNames)
+foreach (_variableName ${_variableNames})
+    message(STATUS "${_variableName}=${${_variableName}}")
```

Comment:
Do you really want to include that, or was this just to debug while you were working on it ?

Reply:
No. You are right what I really wanted to do is to log all the variables in config.log.
This is fixed. 
But probably would be better included in the jrl-cmakemodules.